src/rpc/core_rpc_server.cpp

```diff
@@ -1209,6 +1209,43 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_generateblocks(const COMMAND_RPC_GENERATEBLOCKS::request& req, COMMAND_RPC_GENERATEBLOCKS::response& res, epee::json_rpc::error& error_resp)
+  {
+    PERF_TIMER(on_generateblocks);
+
+    CHECK_CORE_READY();
+    
+    if(m_core.get_nettype() != FAKECHAIN)
+    {
+      error_resp.code = CORE_RPC_ERROR_CODE_REGTEST_REQUIRED;
+      error_resp.message = "Regtest required when generating blocks";      
+      return false;
+    }
+
+    COMMAND_RPC_GETBLOCKTEMPLATE::request template_req;
+    COMMAND_RPC_GETBLOCKTEMPLATE::response template_res;
+    COMMAND_RPC_SUBMITBLOCK::request submit_req;
+    COMMAND_RPC_SUBMITBLOCK::response submit_res;
+
+    template_req.reserve_size = 1;
+    template_req.wallet_address = req.wallet_address;
+    submit_req.push_back(boost::value_initialized<std::string>()); 
+
+    for(size_t i = 0; i < req.amount_of_blocks; i++)
+    {
+    
+      if(!on_getblocktemplate(template_req, template_res, error_resp)) return false;
+
+      submit_req.front() = template_res.blocktemplate_blob;
+      if(!on_submitblock(submit_req, submit_res, error_resp)) return false;
+
+      res.height = template_res.height;
+      res.status = submit_res.status;
+    }
```

Comment:
if req.amount_of_blocks is 0, res.status is uninitialized.
The interetsing case where you might want to carry it over is when a RPC returns false anyway, not true.

Reply:
Added proper initialization for height, and added more moments where status is updated